### ISSUE TITLE
General Layout

### ISSUE THREAD

**Issue by lychrel** [lychrel/pycml#2]
The most common use-cases are:
1. user wants to see visualization/etc. (education)
2. user wants to plug their mappings or coupling schemes into a general architecture (research)

**Comment by lychrel** [comment]
Example generalized class (with the help of class encapsulations):

```python
from maps import KanekoLogistic
from couplings import TwoNeighbor

cml = CML(size=100, map=KanekoLogistic(alpha=1.4), coupling=TwoNeighbor(strength=0.5))
```

This would greatly enhance usability with minimal refactoring. With the next commit (the Evolution class), LogisticMap will be a half-finished generalized CML class. Module partitions as implied above will happen in the next commit after Evolution.

**Comment by lychrel** [comment]
(Taking some cues from how Keras handles this sort of design)

**Comment by lychrel** [comment]
Implemented in de0b12ad1b3dacf9bb12fa025929dd865b4a55d9

**Comment by lychrel** [comment]
Implemented in de0b12ad1b3dacf9bb12fa025929dd865b4a55d9
